src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -87,3 +119,303 @@ export const validateStorageSubmissionParams = celebrate({
     version: Joi.number().required(),
   }),
 })
+
+export const createFormsgReqBody: SharedSubmissionMiddlewareHandlerType =
+  async (req, res, next) => {
+    if (req.body.formsg) return res.send(new FormsgReqBodyExistsError())
+    else {
+      req.body.formsg = {}
+      return next()
+    }
+  }
+
+export const retrieveForm: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const { formId } = req.params
+
+  const logMeta = {
+    action: 'retrieveForm',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  // Retrieve form
+  const formResult = await FormService.retrieveFullFormById(formId)
+  if (formResult.isErr()) {
+    logger.warn({
+      message: 'Failed to retrieve form from database',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  req.body.formsg.formDef = formResult.value
+  return next()
+}
+
+export const checkPublicKey: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const formDef = req.body.formsg.formDef
+
+  if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+  const logMeta = {
+    action: 'checkPublicKey',
+    ...createReqMeta(req),
+    formId: formDef.id,
+  }
+
+  // Retrieve public key.
+  const publicKey = formDef.publicKey
+
+  if (!publicKey) {
+    logger.warn({
+      message: 'Form does not have a public key',
+      meta: logMeta,
+    })
+    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Form does not have a public key',
+    })
+  }
+
+  return next()
+}
+
+export const checkEncryptMode: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const formDef = req.body.formsg.formDef
+
+  if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+  const logMeta = {
+    action: 'checkEncryptMode',
+    ...createReqMeta(req),
+    formId: formDef.id,
+  }
+
+  const checkFormIsEncryptModeResult = checkFormIsEncryptMode(formDef)
+  if (checkFormIsEncryptModeResult.isErr()) {
+    logger.error({
+      message:
+        'Trying to submit non-encrypt mode submission on encrypt-form submission endpoint',
+      meta: logMeta,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      checkFormIsEncryptModeResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  req.body.formsg.encryptedFormDef = checkFormIsEncryptModeResult.value
```

Comment:
same here, may want to store in res.locals

Reply:
Discussed above in [this comment](https://github.com/opengovsg/FormSG/pull/6595#discussion_r1298074277) - conclusion is to leave it as it is 👍